packages/aws-cdk/lib/util/asset-publishing.ts

```diff
@@ -33,15 +40,18 @@ class PublishingAws implements cdk_assets.IAws {
   }
 
   public async discoverDefaultRegion(): Promise<string> {
-    return this.aws.defaultRegion;
+    return this.targetEnv.region;
   }
 
   public async discoverCurrentAccount(): Promise<cdk_assets.Account> {
+    // Try to discover the current partition given default credentials. If we
+    // don't have "default" credentials, (such as when we're using a
+    // CredentialProvider), then we'll just guess the partition to be 'aws'. Our
+    // credential provider protocol currently doesn't provide for partition
+    // information.
     const account = await this.aws.defaultAccount();
```

Comment:
Shouldn't that be based on `this.targetEnv.account` instead of `defaultAccount`?

Reply:
It should. I was making a simplification with a plan to fix it later. But we could just take the current configuration to look up the partition anyway.

It'll be less efficient but more correct.